### DISCUSSION TITLE
Bootstrap - Containers and Grid System - LitReads

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/mern#239] [Tasks]
This business problem scenario which uses Bootstrap 4, HTML, and CSS and involves several steps. 

### Scenario
**Business Problem:** Develop a responsive web page for a fictional book store named "LitReads". The page should showcase various book categories, a list of featured books, and a newsletter subscription form.

### Objectives
1. **Responsive Layout:** Use Bootstrap's grid system to create a responsive layout.
2. **HTML Forms:** Implement a newsletter subscription form.
3. **Semantic HTML:** Utilize semantic tags for better structure and SEO.
4. **CSS Styling:** Apply CSS selectors and box-model concepts.
5. **Bootstrap Typography:** Enhance the visual appeal using Bootstrap typography classes.

### Expected Outcomes
- A responsive webpage that adjusts to different screen sizes.
- Clear and accessible form for newsletter subscription.
- Effective use of semantic HTML tags for structure and clarity.
- Aesthetic appeal and readability using CSS and Bootstrap typography.

#### Step 1: Bootstrap Grid System
- **Task:** Create a grid layout with a header, main content area, and footer.
- **Hints:**
  - Use `<div>` elements with classes `container`, `row`, and `col-md-*`.
  - Example: `<div class="container"><div class="row"><div class="col-md-8">...</div><div class="col-md-4">...</div></div></div>`

#### Step 2: HTML Forms
- **Task:** Implement a newsletter subscription form in the footer.
- **Hints:**
  - Use the `<form>` tag with input fields for name and email.
  - Example: `<form><input type="text" name="name" placeholder="Name"></input>...</form>`

#### Step 3: Semantic HTML
- **Task:** Use semantic tags like `<header>`, `<main>`, `<section>`, `<article>`, and `<footer>`.
- **Hints:**
  - Replace non-semantic `<div>` tags with appropriate semantic tags.
  - Example: `<header>...</header><main>...</main><footer>...</footer>`

#### Step 4: CSS Styling
- **Task:** Style the elements using CSS for margins, padding, borders, and colors.
- **Hints:**
  - Target elements using CSS selectors.
  - Example: `.newsletter-form { padding: 20px; border: 1px solid #ccc; }`

#### Step 5: Bootstrap Typography
- **Task:** Utilize Bootstrap typography classes to style headings, paragraphs, and lists.
- **Hints:**
  - Use classes like `text-primary`, `font-weight-bold`, `lead`.
  - Example: `<h1 class="display-4 text-primary">Welcome to LitReads</h1>`

### Code Snippets (Fill in the Blanks)
```html
<!DOCTYPE html>
<html>
<head>
    <title>LitReads Book Store</title>
    <link rel="stylesheet" href="path/to/bootstrap.css">
    <style>
        .header { /* CSS styles */ }
        .book-list { /* CSS styles */ }
        .footer { /* CSS styles */ }
    </style>
</head>
<body>
    <header class="___1___">
        <h1 class="___2___">LitReads</h1>
    </header>
    <main class="container">
        <section class="row">
            <article class="col-md-8">
                <!-- Book categories -->
            </article>
            <aside class="col-md-4">
                <!-- Featured books -->
            </aside>
        </section>
    </main>
    <footer class="___3___">
        <form class="newsletter-form">
            <input type="___4___" name="email" placeholder="Your Email">
            <button type="submit" class="btn ___5___">Subscribe</button>
        </form>
    </footer>
</body>
</html>
```
**Fill in the Blanks:**
1. `___1___`: "header-class"
2. `___2___`: "display-4 text-primary"
3. `___3___`: "footer-class"
4. `___4___`: "email"
5. `___5___`: "btn-primary"

### Explanation
- The exercise guides candidates through creating a responsive layout using Bootstrap's grid system, implementing a form, and applying styling with CSS. The fill-in-the-blanks approach encourages hands-on learning while providing guidance and structure. This comprehensive exercise should help solidify their understanding of Bootstrap, HTML, and CSS in a practical context.

**Comment by priyeranjan** [comment]
<img width="957" alt="image" src="https://github.com/akash-coded/mern/assets/7481567/31387e98-6151-4123-ba63-0214d5301ede">

**Comment by priyeranjan** [comment]
<img width="809" alt="image" src="https://github.com/akash-coded/mern/assets/7481567/ab1bf035-16a7-442f-b2e1-b4a765e43621">

**Comment by vpervela** [comment]
<img width="1425" alt="Screenshot 2023-12-16 at 7 57 18 AM" src="https://github.com/akash-coded/mern/assets/101532142/11137158-a5c6-4a05-a94c-d64286c74042">

**Comment by Umarani1711** [comment]
![image](https://github.com/akash-coded/mern/assets/78922312/ae0d8a8a-ab12-4ba3-92d1-f549827afad9)

**Comment by arvid-mvo** [comment]
![discussion_239_submit1](https://github.com/akash-coded/mern/assets/101291172/10243c7f-5588-415c-b79b-0af4f1436afb)

**Reply by arvid-mvo** [reply]
![discussion_239_submit2](https://github.com/akash-coded/mern/assets/101291172/e1755b9c-cae4-4d9b-b925-3538de61f30a)

**Comment by Dhanalakshmi-S-23** [comment]
![litsread_op](https://github.com/akash-coded/mern/assets/151384783/bca7a418-1ab2-4a17-affc-3056ca487ab4)

**Comment by AmanBharti07** [comment]
![image](https://github.com/akash-coded/mern/assets/143952663/9165e7e7-2330-4c1d-bc3f-a08a3ce58e9c)
![image](https://github.com/akash-coded/mern/assets/143952663/a04f42f0-9d97-4aff-923a-072cbe1e4397)

**Comment by ujjwal432** [comment]
<img width="1079" alt="Screenshot 2023-12-24 174722" src="https://github.com/akash-coded/mern/assets/61551049/b5202663-d955-463a-8c27-e05c8bef91c0">
